dhis-2/dhis-api/src/main/java/org/hisp/dhis/program/AnalyticsPeriodBoundary.java

```diff
@@ -259,6 +260,7 @@ public void setOffsetPeriodType( PeriodType offsetPeriodType )
     
     @JsonProperty
     @JacksonXmlProperty( namespace = DxfNamespaces.DXF_2_0 )
+    @PropertyRange( min = Integer.MIN_VALUE , max = Integer.MAX_VALUE )
```

Comment:
I guess this could in theory be handled generically in the importer, where we inspect the data type and ensure numeric values are within the min and max limits (like Integer, Long) @mortenoh .

Reply:
@larshelge @mortenoh 
 We have this code in `AbstractPropertyIntrospectorService` for a long time, I'm just afraid if I change the min value here, it may affect other parts of the application.

                Column column = (Column) hibernateProperty.getColumnIterator().next();
                property.setUnique( column.isUnique() );
                property.setRequired( !column.isNullable() );
                property.setMin( 0d );
                property.setMax( (double) column.getLength() );
                property.setLength( column.getLength() );
                if (type instanceof TextType)
                {
                    property.setMin( 0d );
                    property.setMax( (double) Integer.MAX_VALUE );
                    property.setLength( Integer.MAX_VALUE );
                }
                else if (type instanceof IntegerType)
                {
                    property.setMin( 0d );
                    property.setMax( (double) Integer.MAX_VALUE );
                    property.setLength( Integer.MAX_VALUE );
                }
                else if (type instanceof LongType)
                {
                    property.setMin( 0d );
                    property.setMax( (double) Long.MAX_VALUE );
                    property.setLength( Integer.MAX_VALUE );
                }
                else if (type instanceof DoubleType)
                {
                    property.setMin( 0d );
                    property.setMax( Double.MAX_VALUE );
                    property.setLength( Integer.MAX_VALUE );
                }
            }`

Integer validator in `DefaultSchemaValidator`

        if ( !Integer.class.isInstance( propertyObject ) )
        {
            return errorReports;
        }

        Integer value = (Integer) propertyObject;

        if ( !GenericValidator.isInRange( value, property.getMin(), property.getMax() ) )
        {
            errorReports.add( new ErrorReport( klass, ErrorCode.E4008, property.getName(), property.getMin(), property.getMax(), value )
                .setErrorKlass( property.getKlass() ).setErrorProperty( property.getName() ) );
        }

        return errorReports;
    }

Reply:
I added a PR to fix all min values https://github.com/dhis2/dhis2-core/pull/4941